### ISSUE TITLE
Group commit messages by packages in changelog for monorepo

### ISSUE THREAD

**Issue by myndelx** [intuit/auto#1586]
**Is your feature request related to a problem? Please describe.**

Currently, if I merge two PRs related to the same packages and then trigger a release, my changelog will look like this:

```
# v1.0.0 (Wed Oct 14 2020)

#### 🐛 Bug Fix

- `@scope/package-a`
  - This is my first commit message #1 (@author)
- `@scope/package-a`
  - This is my second commit message #2 (@author)
```

**Describe the solution you'd like**

I'd like to have the commit messages to be grouped by packages, like this:

```
# v1.0.0 (Wed Oct 14 2020)

#### 🐛 Bug Fix

- `@scope/package-a`
  - This is my first commit message #1 (@author)
  - This is my second commit message #2 (@author)
```

It should also work for multiple packages:

```
# v1.0.0 (Wed Oct 14 2020)

#### 🐛 Bug Fix

- `@scope/package-a`
  - This is my first commit message #1 (@author)
  - This is my second commit message #2 (@author)
- `@scope/package-a`, `@scope/package-b`
  - This is my third commit message #3 (@author)
```

**Comment by hipstersmoothie** [comment]
This would be a great improvement! no idea how hard it would be but [here is where that code lives](https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L709)

The hard part is that particular code runs on a line by line basis. I can take a look at this one later this week.

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v9.59.0`](https://github.com/intuit/auto/releases/tag/v9.59.0) :rocket:
